das/worker.go

```diff
@@ -18,6 +18,11 @@ import (
 type worker struct {
 	lock  sync.Mutex
 	state workerState
+
+	getter    libhead.Getter[*header.ExtendedHeader]
+	sampleFn  sampleFn
+	broadcast shrexsub.BroadcastFn
+	metrics   *metrics
```

Comment:
I don't understand need for this change in this PR

Reply:
The function is splitted and dependancies needs to be accessed from multiple methods. Passing the whole pack every time looks ugly